### PR TITLE
Mohammad's Trial Project

### PR DESCRIPTION
**Pull request by phpfour**
Hi @A5hleyRich 👋🏼

Here is the PR for the trial project as requested by Brad.

### CODE REVIEW

**Review by A5hleyRich — COMMENTED** [review]
Thanks for taking the time to submit your trial project. I've reviewed this PR like I would any other member of the team so that you get a feel for our code reviews. Overall looks good, just a couple of comments.

File: app/Repositories/Transactions.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    /**
+     * @todo Implement pagination
+     *
+     * @return array
+     */
+    public function getRecentGrouped(): array
+    {
+        $user = Auth::user();
+
+        $transactions = Transaction::query()
```
[the remark this answers is not in the document]
Comment (phpfour) [reply app/Repositories/Transactions.php:21]: No, the `query()` method is not necessary here actually...I've seen many of the active members in the community use it so I guess I picked it up at some point without noticing :)

File: app/Repositories/Transactions.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            ->where('user_id', $user->id)
+            ->orderBy('occurred_at', 'desc')
+            ->get();
```
[the remark this answers is not in the document]
Comment (phpfour) [reply app/Repositories/Transactions.php:24]: Yeah, that would have been a cleaner implementation...like I did it in the `getCurrentBalance()` method: 

```php
return $user->transactions->sum('amount');
```

File: app/Repositories/Transactions.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    public function create(User $user, array $data): Transaction
+    {
+        $transaction = new Transaction();
+
+        $transaction->user_id = Auth::user()->id;
+        $transaction->label = $data['label'];
+        $transaction->amount = $data['amount'];
+        $transaction->occurred_at = Carbon::createFromTimeString($data['occurred_at'])->format('Y-m-d H:i:s');
+
+        $transaction->save();
```
[the remark this answers is not in the document]
Comment (phpfour) [reply app/Repositories/Transactions.php:69]: I usually avoid all kinds of mass assignments and opt-in for explicitly as much as possible. This reduces the security-related issues as well as makes the code clear & obvious for the reader. 

Agree on the mutator and the saving through the user relation 👍🏼

File: app/Repositories/Transactions.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    public function update(Transaction $transaction, array $data): Transaction
+    {
+        $transaction->label = $data['label'];
+        $transaction->amount = $data['amount'];
+        $transaction->occurred_at = Carbon::createFromTimeString($data['occurred_at'])->format('Y-m-d H:i:s');
+
+        $transaction->save();
+
+        return $transaction;
```
[the remark this answers is not in the document]
Comment (phpfour) [reply app/Repositories/Transactions.php:82]: Same reason as above

File: app/Http/Controllers/TransactionController.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        $transactions->create($request->user(), $request->validated());
+
+        $request->session()->flash('status', 'Balance entry saved successfully.');
+
+        return redirect()->route('dashboard');
+    }
+
+    public function update(Transaction $transaction, TransactionStoreRequest $request, Transactions $transactions)
+    {
+        $transactions->update($transaction, $request->validated());
```
[the remark this answers is not in the document]
Comment (phpfour) [reply app/Http/Controllers/TransactionController.php:23]: I'd usually add the [Laratrust](https://laratrust.santigarcor.me/) package for maintaining ACL, so didn't consider that here. 

However, if I'd do without it then probably the `TransactionStoreRequest` class would be a better place for adding the validation.

File: app/Http/Controllers/TransactionController.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+        return response('OK');
+    }
+
+    public function destroy(Transaction $transaction, Transactions $transactions)
+    {
+        $transactions->delete($transaction);
```
[the remark this answers is not in the document]
Comment (phpfour) [reply app/Http/Controllers/TransactionController.php:30]: Same as above

**Comment by phpfour** [comment]
@A5hleyRich Many thanks for your feedback. Some of the points I've made above are personal preferences, which might not hold when working with a team as the consensus among the team members is more important than any individual's personal preference. 

To ensure there is common ground between team members, I thus try to opt-in for a mutually agreed standard like [this](https://spatie.be/guidelines/laravel-php) whenever possible.

**Pull request closed without merging**